cockatrice/src/game/player/player.cpp

```diff
@@ -3444,9 +3457,11 @@ void Player::actPlay()
 
 void Player::actHide()
 {
-    auto *card = game->getActiveCard();
-    if (card) {
-        card->getZone()->removeCard(card);
+    for (const auto &item : scene()->selectedItems()) {
+        auto *card = static_cast<CardItem *>(item);
+        if (card && isUnwritableRevealZone(card->getZone())) {
```

Comment:
actHide would only exist on an unwritable reveal zone, we didn't check for this before, why do we need to check for it now?

Reply:
https://github.com/user-attachments/assets/faa7361d-bd72-4f52-a0b0-ddc97623f20a

If you select cards across multiple zones, then you literally get to nuke cards from existence (client-side), causing really weird desyncs if you do it in a multiplayer game.

Reply:
oh, I didn't know it allowed selections across zones, good catch!

Reply:
could we maybe add a little check at the end here that closes the view if it is empty after?

Reply:
I can't think of a good reason to keep around a view after all of its cards have been removed

Reply:
sure, i'll add that in a separate PR